File: Microservices/AuthenticationAPI/AuthenticationAPI/DTOs/CreateOrUpdateUserDTO.cs
namespace AuthenticationAPI.DTOs
{
    public class CreateOrUpdateUserDTO
    {
        public required string Name { get; set; }
        public required string Document { get; set; }
        public required DateOnly Birthdate { get; set; }
        public required string Email { get; set; }
        public required string Password { get; set; }
        public required string SecretQuestion { get; set; }
        public required string SecretAnswer { get; set; }
    }
}

File: src/front-web/src/services/authentication/dtos/CreateOrUpdateUserDto.js

export default class CreateOrUpdateUserDto {
    constructor(name, email, password, document, birthdate, secretQuestion, secretAnswer) {
      this.name = name;
      this.email = email;
      this.password = password;
      this.document = document;
      this.birthdate = birthdate;
      this.secretQuestion = secretQuestion;
      this.secretAnswer = secretAnswer;
      console.log("CreateOrUpdateUserDto -> Construtor completo.");
    }

    static CreateDtoToUpdate(email, password, document, secretAnswer){
      return new CreateOrUpdateUserDto('', email, password, document, '0001-01-01', '', secretAnswer);
    }
  }